### PR TITLE
Allow finders to be defined for each action.

### PR DESCRIPTION
**Pull request by parndt**
Per #3119 this will allow one to define a method such as:

```ruby
class PagesController
  module Finders
    class AboutUs
      def call(params)
        # logic for finding a page
      end
    end
  end
end
```

The advantage of this is that it allows any number of custom actions and somewhat removes the hardcoded nature of home and show.

Interested to see the results of CI for this change.

* [x] Needs tests

### CODE REVIEW

File: pages/app/controllers/refinery/pages_controller.rb [hunk 1]
```diff
@@ -3,7 +3,8 @@ class PagesController < ::ApplicationController
     include Pages::RenderOptions
 
     before_action :find_page, :set_canonical
-    before_action :error_404, :unless => :current_user_can_view_page?
+    before_action :error_404, if: :action_has_page_finder?,
```
Comment (gwagener) [inline pages/app/controllers/refinery/pages_controller.rb]: 404 _if_ there's a page finder?

Comment (parndt) [inline pages/app/controllers/refinery/pages_controller.rb]: This is a bit strange. I wonder how this is supposed to work :smile:

**Comment by bricesanchez** [comment]
@parndt Are you able to test this feature for the milestone 3.0.4 ?

**Comment by bricesanchez** [comment]
🆙 @parndt :)

**Comment by parndt** [comment]
@bricesanchez it's on my list, but time is lacking.

**Comment by parndt** [comment]
I'm actually a little puzzled about how to test this properly.

**Comment by parndt** [comment]
Turns out the tests we have already adequately cover this

**Comment by parndt** [comment]
@bricesanchez good to go

**Pull request merged**